### PR TITLE
HW36 Viktar

### PR DESCRIPTION
**Pull request by Amicronpersej8**

### CODE REVIEW

**Review by jathpr — CHANGES_REQUESTED** [review]
plz, fix git issues
you have 9 commits

File: Homework_36.ts [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+					production: "1492 Pictures, Heyday Films, Warner Brothers",
+				}
+    ]
+
+    //1. Собрать в массив все жанры фильмов (без повторения)
+
+    const getGenre = (array: Movies[]) => {
+        const genre = array.map(movie=> movie.genre)
+        const arrayGenres = []
+        return Array.from(new Set(arrayGenres.concat(...genre)))
```
Comment (jathpr) [inline Homework_36.ts:121]: can be done like this
`return Array.from(new Set([].concat(...genre)));`

File: Homework_36.ts [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+    const getActors = (array: Movies[]) => {
+        const actors = array.map(movie=> movie.actors)
+        const arrayActors = []
+        return Array.from(new Set(arrayActors.concat(...actors)))
+    }
+    console.log('All actors', getActors(movies))
+
+    //3. Отсортировать фильмы по рейтингу по убыванию
+
+    const sortRating = (array: Movies[]) =>array.sort((a, b)=> b.imdbRating - a.imdbRating)
```
Comment (jathpr) [inline Homework_36.ts:136]: `sort` mutate the original array
you can write `[...array].sort`

File: Homework_36.ts [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+    console.log('sorted by rating', sortRating(movies))
+
+    //4. Создать новый массив, где объекты фильмов будут состоять из следующих полей: id, title, released, plot
+
+    const getNewArrayMovies = (array: Movies[]) =>array.map(movie=>({id: movie.id, title: movie.title, released: movie.released, plot: movie.plot}))
```
Comment (jathpr) [inline Homework_36.ts:141]: super!
my favorite way to do it
`movies.map(({id, title, released, plot})=>({id, title, released, plot}))`

File: Homework_36.ts [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+    //5. Создать функцию, которая бы принимала массив фильмов и число. А результатом этой функции должен быть отфильтрованный массив,
+    // с фильмами где число равно году выхода фильма.
+
+    const getMovieYear = (array: Movies[], year: number) => array.filter(movie => movie.year === year)
+    console.log(getMovieYear(movies, 2021))
+
+    // 6. Создать функцию, которая бы принимала массив фильмов и строку. А результатом этой функции должен быть новый
+    // отфильтрованный массив, с фильмами, где строка входит в название фильма.
+
+    const getMovieWithString = (array: Movies[], string: string) => array.filter(movie => movie.title.includes(string) === true)
```
Comment (jathpr) [inline Homework_36.ts:153]: you can remove `=== true` part

Comment (jathpr) [reply Homework_36.ts:153]: also - usually search is case insensitive
but here - as you wish

File: Homework_36.ts [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+    console.log('6 task', getMovieWithString(movies, 'Harry Potter and'))
+
+    // 7. Создать функцию, которая бы принимала массив фильмов и строку. А результатом этой функции должен быть
+    // отфильтрованный массив, с фильмами где строка входит в название фильма или в его сюжет.
+    const getMovieWithStringInTitleOrPlot = (array: Movies[], string: string) => {
+        return Array.from(new Set(array.filter(movie => (movie.title.includes(string) === true) || (movie.plot.includes(string) === true))))
```
Comment (jathpr) [inline Homework_36.ts:159]: why do you need `Set` here?

File: Homework_36.ts [hunk 6]
```diff
@@ -0,0 +1,169 @@
...
+    }
+    console.log('7 task', getMovieWithStringInTitleOrPlot(movies, ''))
+
+    // 8. Создать функцию, которая бы принимала 3 параметра: 1)массив фильмов , 2) строка(название поля, например 'title') и строку/число(значение поля "Black Widow"). А результатом
+// этой функции должен быть отфильтрованный массив, где параметры 2 и 3 равны в объекте фильма. Например: передаем (films, 'title', 'Black Widow') и на выходе получаем
+// фильм с id=1 если передаем (films, 'year', 2011) , то получаем фильм с id=2
+
+    const getThisArray = (array: Movies[], string: string, parameter: string|number) => array.filter(movie=>movie[string] === parameter)
```
Comment (jathpr) [inline Homework_36.ts:167]: string: keyof Movies

File: Homework_36.ts [hunk 7]
```diff
@@ -0,0 +1,169 @@
+type Movies = {
```
Comment (jathpr) [inline Homework_36.ts:1]: type Movie
Movies for array

File: Homework_36.ts (hunk 5 above)
Comment (Amicronpersej8) [reply Homework_36.ts:159]: Here it is needed so that there are no duplicates of objects in which there are matches in both title and plot

Comment (jathpr) [reply Homework_36.ts]: it wouldn't work as you expect - as all obj are unique. So {id:1} and {id:1} will both stay in Set
also - it's a bad idea to remove them here. Such a case is a bug in data, you shouldn't fix it on your side.

**Comment by jathpr** [comment]
nice
But you still 6 commits)
And didn't close even one conversation.

**Review by jathpr — APPROVED** [review]

File: Homework_36.ts (hunk 6 above)
Comment (jathpr) [reply Homework_36.ts]: you didn't fix it, but it's cool type - I recommend to try

**Pull request merged**